rackcity/views/it_instance_views.py

```diff
@@ -134,9 +134,9 @@ def instance_add(request):
     if not serializer.is_valid(raise_exception=False):
         failure_message += str(serializer.errors)
 
-    rack_id = data['rack']
-    elevation = data['elevation']
-    height = ITModel.objects.get(id=data['model']).height
+    rack_id = serializer.validated_data['rack'].id
+    elevation = serializer.validated_data['elevation']
+    height = serializer.validated_data['model'].height
```

Comment:
Even though we know the values in data are valid since `serializer.is_valid` passed, if any integer values were passed as strings, they would still be string on `data['blah']` but would have been casted to ints on `validated_data['blah']`. This only matters here because we're using these 3 values for some logic so they need to be ints

Reply:
You

Reply:
Amaze